### PR TITLE
Connect db in both screens

### PR DESCRIPTION
**Pull request by sapirshp**
selections saved between screens
choosing date for meeting saved between screens
isScheduled icons match DB

### CODE REVIEW

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 1]
```diff
@@ -91,8 +96,10 @@ private void buttonSelection(TimeSlot timeSlot) {
 
     void clickedOn(TimeSlot timeSlot, boolean isMemberAmountRefresh) {
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: this funcation is wayyy too long!!
not only the new part, but all of it.

consider breaking it up into 2 and even 3 separate functions

Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: also, i do not understand the meaning of the variable "isMemberAmountRefresh".
can you explain it here, or give it a more clear name?

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 2]
```diff
@@ -116,8 +123,10 @@ void clickedOn(TimeSlot timeSlot, boolean isMemberAmountRefresh) {
 
     void clickedOff(TimeSlot timeSlot) {
         timeSlot.setClicked(false);
+        TimeSlot groupTimeSlot = currentGroup.getTimeSlot(timeSlot);
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: maybe gather lines 125-130 to a separate function?
As it is now, the function is rather long

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 3]
```diff
@@ -183,7 +192,7 @@ void setMembersAmount(int newAmount) {
 
     private int getSelectionNumber(TimeSlot timeSlot) {
         for (TimeSlot slot : slotSelections.keySet()) {
-            if (slot.getDate().equals(timeSlot.getDate()) && slot.getHour().equals(timeSlot.getHour())) {
+            if (slot.getSlotIndex() == timeSlot.getSlotIndex()){
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: i would chache the var names "slot" and "timeSlot", it is not clear what they mean.
maybe "currentTimeSlot" for slots inside the for loop, and "wantedTimeSlot" for function parameter?
or something like that

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 4]
```diff
@@ -199,31 +208,34 @@ private boolean containsInUserClicked(TimeSlot timeSlot){
         return false;
     }
 
-    private void setSelectionNumber(TimeSlot timeSlot, boolean add) {
+    private void setSelectionNumber(TimeSlot timeSlot, TimeSlot groupTimeSlot, boolean add) {
         boolean doneSetting = false;
         for (TimeSlot slot : slotSelections.keySet()) {
-            if (slot.getDate() == timeSlot.getDate() && slot.getHour() == (timeSlot.getHour())) {
+            if (slot.getSlotIndex() == timeSlot.getSlotIndex()) {
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: same here with the variable name "slot" and "timeSlot".

also, function is too long. consider creating a function with the "if" content

File: app/src/main/java/com/example/meetapp/Group.java [hunk 5]
```diff
@@ -31,6 +38,9 @@ public Group(String name, String groupId, String admin, boolean isScheduled) {
         this.admin = admin;
         this.members = new ArrayList<>(Arrays.asList(admin));
         this.isScheduled = isScheduled;
+        this.isFirstEntrance = true;
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/Group.java]: both constructors are basically the same.
i suggest that the second constructor should call the first constructor, and only add the new fields

File: app/src/main/java/com/example/meetapp/Group.java [hunk 6]
```diff
@@ -84,4 +94,33 @@ public String getMembersString() {
...
+        return isFirstEntrance;
+    }
+
+    public HashMap<TimeSlot, Integer> getGroupSlotSelections(){
+        return this.groupSlotSelections;
+    }
+
+    public TimeSlot getTimeSlot(TimeSlot timeSlot){
+        for (TimeSlot ts : groupSlotSelections.keySet()){
+            if (ts.getSlotIndex() == timeSlot.getSlotIndex()){
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/Group.java]: same issue with variable names here

File: app/src/main/java/com/example/meetapp/Group.java [hunk 7]
```diff
@@ -84,4 +94,33 @@ public String getMembersString() {
...
+                return ts;
+            }
+        }
+        return null;
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/Group.java]: why do we return null?
returning null is not a good practice from what i know.
is there a specific reason? or could a meaningfull variable be put here instead

File: app/src/main/java/com/example/meetapp/GroupAdapter.java [hunk 8]
```diff
@@ -58,44 +58,41 @@ public void onClick(View v) {
...
-            case 4:
+            case "Wed":
                 return TextDrawable.builder().buildRound("Wed", Color.BLUE);
-            case 5:
+            case "Thu":
                 return TextDrawable.builder().buildRound("Thu", Color.LTGRAY);
-            case 6:
+            case "Fri":
                 return TextDrawable.builder().buildRound("Fri", Color.GREEN);
-            case 7:
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/GroupAdapter.java]: when i created the function, i debated between leaving saturday as a different case or just to put it as default.

maybe the default should be a clear symbol, what do you think?

File: app/src/main/java/com/example/meetapp/GroupAdapter.java [hunk 9]
```diff
@@ -58,44 +58,41 @@ public void onClick(View v) {
...
-    public int timeInDayFactory(int timeRepr) {
+    public int timeInDayFactory(String timeRepr) {
         switch (timeRepr) {
-            case 1:
+            case "Morning":
                 return R.drawable.morning_icon;
-            case 2:
+            case "Afternoon":
                 return R.drawable.afternoon_icon;
-            case 3:
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/GroupAdapter.java]: same here

File: app/src/main/java/com/example/meetapp/IntentHandler.java [hunk 10]
```diff
@@ -14,6 +14,7 @@
     private final int EDIT_NAME_RESULT_CODE = 2;
     private final int ADD_MEMBERS_RESULT_CODE = 3;
     private final int CHANGE_NAME_AND_MEMBERS = 4;
+    private final int DEFAULT = 5;
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/IntentHandler.java]: i didnt see any other usage of this except for this line:
insideGroupActivity.setResult(DEFAULT, insideGroupIntent);

what is it used for?

File: app/src/main/java/com/example/meetapp/MenuHandler.java [hunk 11]
```diff
@@ -430,16 +432,30 @@ public void onClick(View v) {
...
          meetingChosenDialog.show();
      }
 
+     private void setChosenDate(String chosenDate){
+         String[] dayAndTime = chosenDate.split(" ");
+         String day = dayAndTime[0];
+         if (day.equals("Today")){
+             day = DateSetter.getToday();
+         }
+         dateChosen = String.format("%s %s", day, dayAndTime[1]);
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MenuHandler.java]: you have 2 variables, first is called "chosenDate" and the other "dateChosen ".
change one of them to a better name (and maybe both)

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 12]
```diff
@@ -10,12 +10,27 @@
     private static int availableId = 0;
     private static ArrayList<Group> groups = new ArrayList<>();
 
-    static void createMockSelections(HashMap<TimeSlot, Integer> mockSlotSelections, int membersNum) {
-        for (TimeSlot ts : mockSlotSelections.keySet()){
-            Random rand = new Random();
-            int randomNum = rand.nextInt((membersNum-1 - 0) + 1) + 0;
-            mockSlotSelections.put(ts, randomNum);
+    static HashMap<TimeSlot, Integer> createMockSelections(HashMap<TimeSlot, Integer> mockSlotSelections, int membersNum, Group group) {
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: i think the mock data should stay, because other wise when we show it to people it will just be a blank screen.
So we should keep this, and therefore i will also comment on this

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 13]
```diff
@@ -10,12 +10,27 @@
...
+        if (group.getIsFirstEntrance()) {
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: consider spliiting "if" and "else" statements to seaprate functions

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 14]
```diff
@@ -10,12 +10,27 @@
...
+            for (TimeSlot ts : mockSlotSelections.keySet()) {
+                Random rand = new Random();
+                int randomNum = rand.nextInt((membersNum - 1 - 0) + 1) + 0;
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: why are the "-0" and "+0" necessary??
is it a code syntax?

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 15]
```diff
@@ -10,12 +10,27 @@
...
+                group.getGroupSlotSelections().put(ts,randomNum);
+                group.setIsFirstEntrance(false);
+                mockSlotSelections.put(ts, randomNum);
+            }
+        }else {
+            for (TimeSlot ts : mockSlotSelections.keySet()) {
+                TimeSlot groupTimeSlot = group.getTimeSlot(ts);
+                if (groupTimeSlot != null) {
+                    if (groupTimeSlot.getClicked()){
+                        ts.setClicked(true);
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: change double if to: "if (X && Y)"

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 16]
```diff
@@ -30,10 +45,22 @@ static void buildMockGroups(String userName){
         {
             List<String> members = Arrays.asList("Oren", "Chen", "Sapir");
             Group newGroup = new Group("Group" + i, findNextAvailableId(), userName, members, true);
+            newGroup.setChosenDate(getChosenDate(i));
             addGroupToList(newGroup);
         }
     }
 
+    private static String getChosenDate(int iterationNum){
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: why are the chosen dates hard coded? shouldnt they be according to user's choice?

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 1 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: when you choose new members to add the method will set the right member amount on each clicked slot
so when I call this function it needs to know if its a member amount new setting or if its a regular click

File: app/src/main/java/com/example/meetapp/GroupAdapter.java (hunk 8 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/GroupAdapter.java]: I think clear all symbols is not a function we need now
I think it's ok like this

File: app/src/main/java/com/example/meetapp/GroupAdapter.java (hunk 9 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/GroupAdapter.java]: same

File: app/src/main/java/com/example/meetapp/IntentHandler.java (hunk 10 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/IntentHandler.java]: when beckPressed without changing group name or members it's symply update the group's selections list, so its the default behavior

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 15 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: no because not all the slot will be clicked

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 16 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: its the mock DB, not a user choise

File: app/src/main/java/com/example/meetapp/Group.java (hunk 5 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/Group.java]: I don't want to mess Oren's fireBase code

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 17]
```diff
@@ -69,6 +71,9 @@ void setListeners(Map<String, String> datesToDisplay) {
             int slotIndex = (3 * dateIndex) + hourIndex;
             final TimeSlot timeSlot = new TimeSlot(timeSlotButton, date, hour, slotIndex);
             slotSelections.put(timeSlot, 0);
+            if (currentGroup.getIsFirstEntrance()) {
```
[the remark this answers is not in the document]
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 1 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 2 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed - split to two functions

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 3 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 4 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/Group.java (hunk 6 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/Group.java]: fixed

File: app/src/main/java/com/example/meetapp/Group.java (hunk 7 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/Group.java]: fixed

File: app/src/main/java/com/example/meetapp/MenuHandler.java (hunk 11 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MenuHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 12 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: ok

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 13 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: fixed

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 14 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: you are right, fixed

File: app/src/main/java/com/example/meetapp/Group.java (hunk 5 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/Group.java]: yes, i saw his response...
but i still think that it should be changed.
maybe later on

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java [hunk 18]
```diff
@@ -127,11 +131,15 @@ void clickedOff(TimeSlot timeSlot) {
         groupTimeSlot.setClicked(false);
         userClicks.remove(timeSlot);
         setSelectionNumber(timeSlot, groupTimeSlot, false);
+        setClickedOfBackground(timeSlot);
+    }
+
+    private void setClickedOfBackground(TimeSlot timeSlot){
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: did you mean "setClickedOffBackground"?

File: app/src/main/java/com/example/meetapp/MockDB.java [hunk 19]
```diff
@@ -11,26 +11,34 @@
...
+        for (TimeSlot ts : mockSlotSelections.keySet()) {
+            Random rand = new Random();
+            int randomNum = rand.nextInt((membersNum - 1) + 1);
+            group.getGroupSlotSelections().put(ts,randomNum);
+            group.setIsFirstEntrance(false);
+            mockSlotSelections.put(ts, randomNum);
+        }
+    }
+
+    private static void setGroupSelections(HashMap<TimeSlot, Integer> mockSlotSelections, Group group){
```
Comment (Chen-Ben-yashar) [inline app/src/main/java/com/example/meetapp/MockDB.java]: this function shows to screen the selections, right?
if so, isn't there already a function that does that? why is this needed?

and if not, pls explain what this one does :)

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 16 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/MockDB.java]: got it, so i think the name should be changed.
maybe "set scheduled mock meeting times", or "setGroupsWithScheduledDate".

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 19 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: this function get the group's selection map and display it on the screen.

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 1 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: fixed

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 16 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/MockDB.java]: fixed

File: app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java (hunk 18 above)
Comment (sapirshp) [reply app/src/main/java/com/example/meetapp/CalendarSlotsHandler.java]: yes, fixed

File: app/src/main/java/com/example/meetapp/MockDB.java (hunk 19 above)
Comment (Chen-Ben-yashar) [reply app/src/main/java/com/example/meetapp/MockDB.java]: got it 👍
